pkg/sdk/go/bootstrap.go

```diff
@@ -46,36 +46,41 @@ type ConfigUpdateCertReq struct {
 	CACert     string `json:"ca_cert"`
 }
 
-func (sdk mfSDK) AddBootstrap(token string, cfg BootstrapConfig) (string, error) {
+func (sdk mfSDK) AddBootstrap(token string, cfg BootstrapConfig) (string, errors.SDKError) {
 	data, err := json.Marshal(cfg)
 	if err != nil {
-		return "", err
+		return "", errors.NewSDKError(err.Error())
 	}
 
 	url := fmt.Sprintf("%s/%s", sdk.bootstrapURL, configsEndpoint)
 	req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(data))
 	if err != nil {
-		return "", err
+		return "", errors.NewSDKError(err.Error())
 	}
 
 	resp, err := sdk.sendRequest(req, token, string(CTJSON))
 	if err != nil {
-		return "", err
+		return "", errors.NewSDKError(err.Error())
 	}
 	defer resp.Body.Close()
 
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.NewSDKError(err.Error())
+	}
+
 	if resp.StatusCode != http.StatusCreated && resp.StatusCode != http.StatusOK {
-		return "", errors.Wrap(ErrFailedCreation, errors.New(resp.Status))
+		return "", encodeError(body, resp.StatusCode)
 	}
 
 	id := strings.TrimPrefix(resp.Header.Get("Location"), "/things/configs/")
 	return id, nil
 }
 
-func (sdk mfSDK) Whitelist(token string, cfg BootstrapConfig) error {
+func (sdk mfSDK) Whitelist(token string, cfg BootstrapConfig) errors.SDKError {
 	data, err := json.Marshal(BootstrapConfig{State: cfg.State})
 	if err != nil {
-		return errors.Wrap(ErrFailedWhitelist, err)
+		return errors.NewSDKError(errors.Wrap(ErrFailedWhitelist, err).Error())
```

Comment:
In other SDK methods, we usually have a pattern like this:
```go
data, err := json.Marshal(connIDs)
if err != nil {
    return errors.NewSDKError(err.Error())
}
```
So, without wrapping. Why is this method specific?

Reply:
The function originally returned the wrapped error. So, the outer package of NewSDKError(...) is just to convert from `error` to `SKDError` interface.
Since both ErrFailedWhitelist, and the specific marshalling error, both are required for proper error handling, I have kept both of them. And, ErrWhitelist is used in multiple places, so I kept it.

Reply:
Should I remove ErrWhiteList, and remove this wrapping? Since, `Whitelist` in bootstrap, is the only function, that has its own error now

Reply:
The thing is that errors such as `ErrFailedWhitelist` or `ErrFailed<MethodName>` make no sense here. If the method is `Whitelist`, and it returns an error -  there is not much point in having it wrapped t something as trivial as "failed to whitelist". That's self-evident. The caller, however, can do a wrap to indicate the actual point of failure, but that's not the case here.

Reply:
> The thing is that errors such as `ErrFailedWhitelist` or `ErrFailed<MethodName>` make no sense here. If the method is `Whitelist`, and it returns an error - there is not much point in having it wrapped t something as trivial as "failed to whitelist". That's self-evident. The caller, however, can do a wrap to indicate the actual point of failure, but that's not the case here.

By indicating the actual point of failure like error `ErrFailedWhitelist` , will helps to make decision in automations,  by checking the error contains `ErrFailedWhitelist` automation program can take decisions.

Reply:
Yes, like @arvindh123 mentioned. This specific error `errwhitelist` is used in provision/service also. That's why I'm not removing it.